client/allocrunner/taskrunner/lifecycle.go

```diff
@@ -32,13 +31,17 @@ func (tr *TaskRunner) Restart(ctx context.Context, event *structs.TaskEvent, fai
 	}
 
 	// Drain the wait channel or wait for the request context to be canceled
-	result.Wait(ctx)
+	waitCh, err := handle.WaitCh(ctx)
+	if err != nil {
+		tr.logger.Error("failed to kill task. Resources may have been leaked", "error", err)
```

Comment:
If there is an error will the waitCh not block forever?

Reply:
I think it could, but a context is passed and the doc says:
> Blocks until exiting task exits **or passed-in context is canceled**